dockerfile-image-update/src/test/java/com/salesforce/dockerfileimageupdate/utils/PullRequestsTest.java

```diff
@@ -150,6 +171,158 @@ public void testPullRequestsPrepareSkipsSendingPRIfRepoOnboardedToRenovate() thr
+        HttpGet httpGet = mock(HttpGet.class);
+        GithubAppCheck.CustomException customException = mock(GithubAppCheck.CustomException.class);
+
+        doNothing().when(githubAppCheck).refreshJwtIfNeeded(any(), any());
+        when(httpClient.execute(any(HttpGet.class))).thenReturn(closeableHttpResponse);
+        when(closeableHttpResponse.getStatusLine()).thenReturn(statusline);
+        when(statusline.getStatusCode()).thenReturn(500);
+
+        assertThrows(GithubAppCheck.CustomException.class, () -> githubAppCheck.isGithubAppEnabledOnRepositoryWithGitApi(fullRepoName, httpClient));
+    }
+
+    @Test
+    public void testIsGithubAppEnabledOnRepository_RenovateRetry() throws IOException {
+        String fullRepoName = "org/repo";
+        CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
+        CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
+        GithubAppCheck githubAppCheck = spy(new GithubAppCheck(mock(Namespace.class)));
+        StatusLine statusline = mock(StatusLine.class);
+        HttpGet httpGet = mock(HttpGet.class);
+        Supplier supplier = mock(Supplier.class);
+
+        when(httpClient.execute(any(HttpGet.class))).thenReturn(closeableHttpResponse);
+        when(closeableHttpResponse.getStatusLine()).thenReturn(statusline);
+        when(statusline.getStatusCode()).thenReturn(500);
+        doThrow(new GithubAppCheck.CustomException()).when(githubAppCheck).isGithubAppEnabledOnRepositoryWithRenovateApi(fullRepoName);
+
+        assertThrows(GithubAppCheck.CustomException.class, () -> githubAppCheck.isGithubAppEnabledOnRepository(fullRepoName));
+        verify(githubAppCheck, times(2)).isGithubAppEnabledOnRepositoryWithRenovateApi(fullRepoName);
+    }
+
+    @Test
+    public void testIsGithubAppEnabledOnRepository_GitRetry() throws IOException {
+        String fullRepoName = "org/repo";
+        CloseableHttpClient httpClient = mock(CloseableHttpClient.class);
+        CloseableHttpResponse closeableHttpResponse = mock(CloseableHttpResponse.class);
+        GithubAppCheck githubAppCheck = spy(new GithubAppCheck(mock(Namespace.class)));
+        StatusLine statusline = mock(StatusLine.class);
+        HttpGet httpGet = mock(HttpGet.class);
+        Supplier supplier = mock(Supplier.class);
+
+        doNothing().when(githubAppCheck).refreshJwtIfNeeded(any(), any());
+        when(httpClient.execute(any(HttpGet.class))).thenReturn(closeableHttpResponse);
+        when(closeableHttpResponse.getStatusLine()).thenReturn(statusline);
+        when(statusline.getStatusCode()).thenReturn(500);
+        doThrow(new GithubAppCheck.CustomException()).when(githubAppCheck).isGithubAppEnabledOnRepositoryWithGitApi(fullRepoName);
+
+        assertThrows(GithubAppCheck.CustomException.class, () -> githubAppCheck.isGithubAppEnabledOnRepository(fullRepoName));
+        verify(githubAppCheck, times(2)).isGithubAppEnabledOnRepositoryWithGitApi(fullRepoName); 
```

Comment:
Lets add this verification as well, `verify(githubAppCheck, times(2)).isGithubAppEnabledOnRepositoryWithRenovateApi(fullRepoName);` .  We would expect in this case that both the calls to Renovate and the calls to the Git api have been made in this test.

Reply:
@cjneasbi resolved